extensions/s3/src/main/java/io/deephaven/extensions/s3/S3SeekableChannelProvider.java

```diff
@@ -49,27 +51,65 @@
  */
 final class S3SeekableChannelProvider implements SeekableChannelsProvider {
 
+    private static final int MAX_KEYS_PER_BATCH = 1000;
+
+    private static final Logger log = LoggerFactory.getLogger(S3SeekableChannelProvider.class);
+
     /**
-     * We always allocate buffers of maximum allowed size for re-usability across reads with different fragment sizes.
-     * There can be a performance penalty though if the fragment size is much smaller than the maximum size.
+     * Cache of shared {@link S3AsyncClient} objects and request caches, keyed by the S3 instructions.
      */
-    private static final BufferPool BUFFER_POOL = new BufferPool(S3Instructions.MAX_FRAGMENT_SIZE);
+    private static final Map<S3Instructions, ClientData> SHARED_CLIENT_DATA = new ConcurrentHashMap<>();
```

Comment:
It looks like this map only ever grows? Do we need some mechanism to remove entries?